tools/device_configuration_checker.py

```diff
@@ -345,4 +349,161 @@ def _run_tests_on_know_good_image(args, device):
         return (result, result_str)
 
     except Exception as error:
+        import traceback
+        traceback.print_exc()
+        logger.error(traceback.format_exc())
         return (False, "Image Test result: " + str(error))
+
+def create_work_directory(directory):
+    """
```

Comment:
The part you mentioned earlier that could go in a config file sort of falls in the same bucket as this.
I agree that it could be a separate task, but what I've seen hints to the need for a bit of refactoring.